LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -1125,7 +1125,7 @@ extension DoseStore {
             case .failure(let error):
                 completion(.failure(.persistenceError(description: error.localizedDescription, recoverySuggestion: nil)))
             case .success(let insulinDeliveryDoses):
-                let filteredStart = max(insulinDeliveryDoses.lastBasalEndDate ?? start, start)
+                let filteredStart = max(self.lastPumpEventsReconciliation ?? start, start)
```

Comment:
Why this change? I checked the code before my large DoseStore update and it used `insulinDeliveryDoses.lastBasalEndDate`, too.

Reply:
This is a fix for a pretty severe issue where Loop displays double IOB at times, if a pump reports reservoir more recently than a finalized basal dose. Mostly an issue for DIY MDT; should not affect pods. See https://github.com/LoopKit/LoopKit/pull/344